.github/workflows/release.yaml

```diff
@@ -0,0 +1,112 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+name: "NeMo Aligner release"
+
+on:
+  workflow_dispatch:
+    inputs:
+      branch: 
+        description: Branch to release
+        required: true
+        type: string
+jobs: 
+  main:
+    if: contains(fromJSON('["terryk", "gshennvm", "ko3n1g"]'), github.actor)
+    runs-on: ubuntu-latest
+    environment: 
+      name: main
+    steps:  
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          path: ${{ github.run_id }}
+          ref: ${{ inputs.branch }}
+
+      - name: Create release
+        id: version-number
+        run: |
+          cd ${{ github.run_id }}
+          VERSION=$(python -c "import nemo_aligner; print(nemo_aligner.__version__)")
+          
+          NAME="NVIDIA NeMo-Aligner ${VERSION}"
+          CHANGELOG=$(awk '/^## '"$NAME"'/{flag=1; next} /^## /{flag=0} flag' CHANGELOG.md)
+          CHANGELOG=$(echo "$CHANGELOG" | sed '/./,$!d' | sed ':a;N;$!ba;s/\n$//')
+
+          PAYLOAD=$(jq \
+                      -n \
+                      -c \
+                      --arg CI_COMMIT_BRANCH "${{ inputs.branch }}" \
+                      --arg NAME "$NAME" \
+                      --arg BODY "$CHANGELOG" \
+                      '{
+                        "tag_name": $CI_COMMIT_BRANCH,
+                        "target_commitish": $CI_COMMIT_BRANCH,
+                        "name": $NAME,
+                        "body": $BODY,
+                        "draft": false,
+                        "prerelease": false,
+                        "generate_release_notes": false
+                      }'
+                  )
+
+          curl -L \
+            -X POST \
+            -H "Accept: application/vnd.github+json" \
+            -H "Authorization: Bearer ${{ secrets.PAT }}" \
+            -H "X-GitHub-Api-Version: 2022-11-28" \
+            https://api.github.com/repos/NVIDIA/NeMo-Aligner/releases \
+            -d "$PAYLOAD"
+      
+      - name: Build, test, and release wheel
+        env:
+          TWINE_USERNAME: ${{ secrets.TWINE_USERNAME }}
+          TWINE_PASSWORD: ${{ secrets.TWINE_PASSWORD }}
+        run: |
+          cd ${{ github.run_id }}
+          EXPECTED_VERSION=$(python -c 'import nemo_aligner; print(nemo_aligner.__version__)')
+
+          python3 -m pip install --upgrade build
+          python3 -m build
+
+          pip install dist/*.whl
+
+          cd ../
+
+          INSTALLED_VERSION=$(python -c 'import nemo_aligner; print(nemo_aligner.__version__)')
+          
+          if [[ "$INSTALLED_VERSION" != "$EXPECTED_VERSION" ]]; then
```

Comment:
This check was meant as a sanity check to make sure that we can install and run the PyPi package (and that it spits out the desired version number). We wanted to do some kind of minimal test before publishing the wheel. 

Open to discuss if we want to remove this check

Reply:
I see. So this check is an e2e installation check, at first it appeared like it was trying to catch mismatches between the version that setuptools is passed vs. what is stored that the root package. In that case, I wonder if it makes sense to drop the `--no-deps` I previously suggested, since I'm inferring that you also wanted to catch `-z INSTALLED_VERSION` since that means the installation got messed up (e.g., b/c of a dep conflict).

The only issue with dropping `--no-deps` is the version of nemo/mcore pulled from pypi isn't the one we store in the dockerfile, so there's some possibility of false positives/negatives.

Reply:
Maybe we should run this verification step inside of the container?
The container resembles the environment that we expect our customer to have. So we should install it maybe inside the expected target environment and check that the installation passed.

Does that make sense?

Reply:
Yea, that's a good idea. I think inside the container makes the most sense